VideoLocker/src/main/java/org/edx/mobile/module/db/IDatabase.java

```diff
@@ -359,5 +359,10 @@ DownloadedState getDownloadedStateForVideoId(String videoId,
      */
     public Boolean isAnyVideoDownloadingInSubSection(DataCallback<Boolean> callback, String courseId,
                                                      String section, String subSection);
-    
+
+    /**
+     *
+     * @param userName
+     */
+    public void setUserName(String userName);
```

Comment:
@aleffert the problem is related to the way we implemented right now.  when the app is first launched, the code try to initial the database , at that time username may not be available yet.